packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
+      dateDispatch(dateState);
+    }
+  }, [momentDateFormat, dateValue, value]);
+
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {number} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateOrder - Date variant based on the currently supported date
+   *                      orders: 'MDY', 'DMY', or 'YMD'.
+   */
+  const moveFocusOnChange = (inputValue, type, dateOrder) => {
+    if (dateOrder === 'MDY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === 'DMY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
+          monthInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === 'YMD') {
+      if (inputValue.length === 4) {
+        monthInputRef.current.focus();
+      } else if (inputValue.length === 2 && type === DateUtil.inputType.MONTH) {
+        dayInputRef.current.focus();
+      }
+    }
+  };
+
+  /**
+   * Sets state for the day, month, and year.
+   * @param {string} inputValue - The input value to set in state.
+   * @param {number} type - The inputType (day, month, or year).
+   */
+  const setDate = (inputValue, type) => {
+    if (type === DateUtil.inputType.DAY) {
+      dateDispatch({ day: inputValue });
+    } else if (type === DateUtil.inputType.MONTH) {
+      dateDispatch({ month: inputValue });
+    } else {
+      dateDispatch({ year: inputValue });
+    }
+
+    moveFocusOnChange(inputValue, type, dateFormatOrder);
+  };
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
+  const handleDateChange = (event, inputValue, type) => {
+    /**
+     * Sets the day, month and year based on input values, formats them
+     * based on the date format variant, and passes the formatted date to onChange.
+     */
+    const day = type === DateUtil.inputType.DAY ? inputValue : date.day;
+    const month = type === DateUtil.inputType.MONTH ? inputValue : date.month;
+    const year = type === DateUtil.inputType.YEAR ? inputValue : date.year;
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(false);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(false);
+    } else {
+      setYearInitialFocused(false);
+    }
+
+    let inputDate;
+    let formattedDate;
+    if (day.length === 2 && month.length === 2 && year.length === 4) {
+      inputDate = DateUtil.convertToISO8601(`${year}-${month}-${day}`, 'YYYY-MM-DD');
```

Comment:
Are we able to use the dateForamtOrder to correctly create this formatting for the formatted date?? Why would it be used at some times and other times we piece a date together?

Reply:
Since in `handleDateChange`, we have the individual input values for day, month and year. I am just piecing it into ISO format (YMD). It can be pieced into any format here but the final value should be an ISO value.